packages/core/client.js

```diff
@@ -114,11 +134,53 @@ class BugsnagClient {
   }
 
   startSession () {
-    if (!this._sessionDelegate) {
-      this._logger.warn('No session implementation is installed')
+    const session = new BugsnagSession()
+    // run synchronous onSession callbacks
+    let ignore = false
+    const cbs = this._cbs.s.slice(0)
+    while (!ignore) {
+      if (!cbs.length) break
+      try {
+        ignore = cbs.pop()(session) === false
+      } catch (e) {
+        this._logger.error('Error occurred in onSession callback, continuing anyway…')
+        this._logger.error(e)
+      }
+    }
```

Comment:
This could be refactored although it doesn't help to reduce the bundle diff:

```
  runSyncCallbacks (callbacks, callbackArg, callbackType) {
    let ignore = false
    const cbs = callbacks.slice(0)
    while (!ignore) {
      if (!cbs.length) break
      try {
        ignore = cbs.pop()(callbackArg) === false
      } catch (e) {
        this._logger.error(`Error occurred in ${callbackType} callback, continuing anyway…`)
        this._logger.error(e)
      }
    }
    return ignore
  }
```

Reply:
I'll take this – we can reuse it for the internal session payload callbacks too.